### PR TITLE
added functions for creating sRGB enabled textures from file / memory, etc.

### PR DESCRIPTION
**Pull request by EnokViking**
..and removed some unused functions, sfTexture_setSrgb and sfTexture_loadFromFile.

### CODE REVIEW

File: include/SFML/Graphics/Texture.h [hunk 1]
```diff
@@ -68,6 +68,31 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_create(unsigned int width, unsigned int
 ////////////////////////////////////////////////////////////
 CSFML_GRAPHICS_API sfTexture* sfTexture_createFromFile(const char* filename, const sfIntRect* area);
 
+////////////////////////////////////////////////////////////
+/// \brief Create a texture configured for srgb from a file
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:72]: sRGB

File: include/SFML/Graphics/Texture.h [hunk 2]
```diff
@@ -68,6 +68,31 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_create(unsigned int width, unsigned int
...
+/// When providing texture data from an image file or memory, it can
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:73]: I think there should be an empty comment line between brief and the extended text

File: include/SFML/Graphics/Texture.h [hunk 3]
```diff
@@ -68,6 +68,31 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_create(unsigned int width, unsigned int
...
+/// either be stored in a linear color space or an sRGB color space.
+/// Most digital images account for gamma correction already, so they
+/// would need to be "uncorrected" back to linear color space before
+/// being processed by the hardware. The hardware can automatically
+/// convert it from the sRGB color space to a linear color space when
+/// it gets sampled. When the rendered image gets output to the final
+/// framebuffer, it gets converted back to sRGB.
+///
+/// After enabling or disabling sRGB conversion, make sure to reload
+/// the texture data in order for the setting to take effect.
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:83]: This doesn't make sense here, as we're already loading the texture

File: include/SFML/Graphics/Texture.h [hunk 4]
```diff
@@ -80,6 +105,18 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_createFromFile(const char* filename, con
 ////////////////////////////////////////////////////////////
 CSFML_GRAPHICS_API sfTexture* sfTexture_createFromMemory(const void* data, size_t sizeInBytes, const sfIntRect* area);
 
+////////////////////////////////////////////////////////////
+/// \brief Create a new texture from a file in memory
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:109]: Maybe squeeze `sRGB` somewhere in

File: include/SFML/Graphics/Texture.h [hunk 5]
```diff
@@ -91,6 +128,17 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_createFromMemory(const void* data, size_
 ////////////////////////////////////////////////////////////
 CSFML_GRAPHICS_API sfTexture* sfTexture_createFromStream(sfInputStream* stream, const sfIntRect* area);
 
+////////////////////////////////////////////////////////////
+/// \brief Create a new texture from a custom stream
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:132]: Maybe squeeze `sRGB` somewhere in

File: include/SFML/Graphics/Texture.h [hunk 6]
```diff
@@ -102,6 +150,17 @@ CSFML_GRAPHICS_API sfTexture* sfTexture_createFromStream(sfInputStream* stream,
 ////////////////////////////////////////////////////////////
 CSFML_GRAPHICS_API sfTexture* sfTexture_createFromImage(const sfImage* image, const sfIntRect* area);
 
+////////////////////////////////////////////////////////////
+/// \brief Create a new texture from an image
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Texture.h:154]: Maybe squeeze `sRGB` somewhere in

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Just some documentation adjustments 🙂

**Review by Bambo-Borris — APPROVED** [review]
Checked it out locally and compared C++ results to C ones and got the same end result.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thanks a ton for getting this fixed! 🙂

**Pull request merged**